src/message/pack_encrypted/mod.rs

```diff
@@ -129,6 +134,72 @@ impl Message {
 
         Ok((msg, metadata))
     }
+
+    fn validate(&self, to: &str, from: Option<&str>, sign_by: Option<&str>) -> Result<bool> {
+        if !is_did(to) {
+            Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`to` value is not a valid DID of DID URL: {}", to),
+            ))?;
+        }
+
+        match from {
+            Some(from) if !is_did(from) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`from` value is not a valid DID of DID URL: {}", from),
+            ))?,
+            _ => {}
+        }
+
+        match sign_by {
+            Some(sign_by) if !is_did(sign_by) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!(
+                    "`sign_from` value is not a valid DID of DID URL: {}",
+                    sign_by
+                ),
+            ))?,
+            _ => {}
+        }
+
+        let (to_did, _) = did_or_url(to);
+
+        let self_to = &self.to;
```

Comment:
Seems If `self.to` is Some and  `to` is None you will not rise an error that i would expect to be rised.

Reply:
`let self_to = &self.to;` i don't see reasons for this variable